### DISCUSSION TITLE
How to have `/year/month/slug.html` as blog URL

### DISCUSSION THREAD

**Discussion by copdips** [squidfunk/mkdocs-material#6430] [Q&A] [answered]
Hello,

Current blog post url format is:
`/blog/year/month/day/slug/`, 

as I'm migrating from other blog system to mkdocs-material, the orignal url format is:
`/year/month/slug.html`,  

is there a way to keep the orignal format in mkdocs-material ? Thanks

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello,
I modified the https://github.com/mkdocs-material/examples/tree/master/examples/blog-basic example for your needs;
[blog-basic-for-copdips.zip](https://github.com/squidfunk/mkdocs-material/files/13547143/blog-basic-for-copdips.zip)

To remove the `blog` prefix of the URL you need to setup a stand-alone blog only site:
https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.blog_dir-blog-only
To remove the `day` part of the URL you need to change the `post_url_date_format` option:
https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.post_url_date_format
To enable `.html` path endings you need to disable the `use_directory_urls` global MkDocs option:
https://www.mkdocs.org/user-guide/configuration/#use_directory_urls

**Reply by kamilkrzyskow** [reply]
@copdips I forgot to restore a file, before sending the example, it's fixed now in the edited comment.

**Reply by copdips** [reply]
Thanks a lot @kamilkrzyskow for your prompt response, the zip file you provided especially for me works like a charm !
Now I have the same URLs between the old blog system and the new mkdocs-material one, so cool.

And I noticed that in the zip, the `posts` folder is under `docs` folder directly which is good for my needs, but the exmaple in the repo is still under `docs/blog` folder:

![image](https://github.com/squidfunk/mkdocs-material/assets/13083856/c7c1d77b-1213-4fad-a81f-39f5a3e6e5f7)

**Reply by kamilkrzyskow** [reply]
Yes, it's inside the `blog` directory, because that's the default for the Documentation + Blog setup:
https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.blog_dir-documentation--blog
But you need to use the Blog only setup with files inside the root of docs to achieve the result you want.

**Reply by copdips** [reply]
thanks for this trick, it would be nice to document this somewhere for other users having the same needs probabally :)

**Reply by kamilkrzyskow** [reply]
But it's documented: https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.blog_dir
Are you saying that the requirement of changing the path via `blog_dir` isn't enough of an indication that the files need to be in the root of `docs` as well?

**Reply by copdips** [reply]
yes, I just made a pull request, please check:
https://github.com/squidfunk/mkdocs-material/pull/6435

**Comment by copdips** [comment]
@kamilkrzyskow 
One more question about slug genetation please.

Say I have a blog post with top level head:

```
# Github - Test
```

Within the old blog system, its slug is : `github-test`, but within mkdocs material, by default, it's `github---test`, is there any way to keep the old way, which means consider `-` as special chars, just like white spaces. And would be better to have a param to set a list of such special chars.

**Reply by squidfunk** [reply]
You can try the standard slug function from Python Markdown, which filters all Unicode characters:

``` yaml
plugins:
  - blog:
      post_slugify: !!python/name:markdown.extensions.toc.slugify
```

However, you'll get an error when you'll do this anywhere except on `master`, because I just discovered that the typings are too strict, which I fixed in adad8238c, so either use `master` or wait for the next release.

Additionally, you might encounter cases where you need to set the [`slug`](https://squidfunk.github.io/mkdocs-material/plugins/blog/#meta.slug) manually. This is also possible and simple:

``` markdown
---
slug: help-im-trapped-in-a-universe-factory
---

# Post title
...
```

**Reply by copdips** [reply]
thx a lot, I'll wait for the new release !
